src/main/node/lib/app.js

```diff
@@ -29,16 +29,16 @@ function makeApp(fn) {
 
         switch (negotiator.mediaType(SUPPORTED_MEDIA_TYPES)) { // returns the most preferred Accept'ed type intersected with our list
             case 'application/json':
-                res.type("application/json").json(resultx);
+                res.json(resultx);
                 break;
             case 'text/plain':
                 // Force text/plain before calling send, as it defaults to html for strings
-                res.type("text/plain").send("" + resultx);
+                res.type('text/plain').send('' + resultx);
+                break;
+            default:
+                res.status(406).type('text/plain').send('' + resultx);
```

Comment:
Won't this create a weird situation where this "seems to work" (the result is produced) but "not quite" (406 code) and some users will not notice right away?
Seems to me we should be either 100% lenient or 100% strict, but not half-half. What do you think?

Reply:
A 406 has the option of either returning a listing of it's supported MIME types, or choose the "most appropriate" type for the response. At this point the function has executed successfully, the only question is how we respond. If a hard fail is desired, we will need to detect that the accepted type is not valid before invoking the function. Since this endpoint is only invoked within riff, it's probably better to be strict and fail fast.

With the previous code path the request will hang open and never receive a response.

Reply:
I see. Didn't even realize that the previous code would hang.
Given that we may drop `http` as the invocation protocol soon anyway, I guess merging your PR as is is ok.